### PR TITLE
Custom privacy request fields

### PR DESCRIPTION
**Pull request by galvana**
#### Privacy Center
An admin can now configure custom fields to collect as part of a privacy request. This is done by configuring the `custom_privacy_request_fields` in the Privacy Center's `config.json`
```
"custom_privacy_request_fields": {
  "first_name": {
    "label": "First name",
    "required": true
  },
  "last_name": {
    "label": "Last name",
    "required": false
  },
  "order_id": {
    "label": "Order ID",
    "required": true
  }
}
```
image

#### Fides
The storage of the data collected from the privacy center can be enabled with the `CONFIG.execution.allow_custom_privacy_request_field_collection` system setting and the data can be included in a privacy request by enabling `CONFIG.execution.allow_custom_privacy_request_fields_in_request_execution`.

#### Admin UI
The privacy request details will now show the custom fields that were collected from the Privacy Center
image

### Code Changes

* [ ] Updated the privacy center to be able to collect custom privacy request fields
* [ ] Updated the privacy request details and privacy request approval modal to include custom privacy request fields
* [ ] Adding the `include_custom_privacy_request_fields` query param to the GET /privacy-request endpoint
* [ ] New `custom_privacy_request_field` table to store the custom fields
* [ ] Updated the `privacy_request` model to read/write custom fields to the cache and database
* [ ] Adding `<custom_privacy_request_fields>` placeholder that is available for access and erasure SaaS requests
* [ ] Two new execution system settings (mentioned in Description Of Changes)
* [ ] Ability so specify dot-delimited paths for SaaS placeholders
* [ ] Ability to specify a placeholder as optional, with the `?` notation (ex: <custom_privacy_request_fields.last_name?>)
* [ ] Misc UI layout fixes

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/privacy-center/components/modals/RequestModal.tsx [hunk 1]
```diff
@@ -11,9 +11,9 @@ const RequestModal: React.FC<RequestModalProps> = ({
   onClose,
   children,
 }) => (
-  <Modal isOpen={isOpen} onClose={onClose}>
+  <Modal isOpen={isOpen} onClose={onClose} isCentered>
     <ModalOverlay />
-    <ModalContent top={[0, "205px"]} maxWidth="464px" mx={5} my={3}>
+    <ModalContent maxWidth="464px" mx={5} my={3}>
```
Comment (Kelsey-Ethyca) [inline clients/privacy-center/components/modals/RequestModal.tsx:16]: We need to set a maxHeight

File: src/fides/api/models/privacy_request.py [hunk 2]
```diff
@@ -924,6 +1011,62 @@ def as_identity_schema(self) -> Identity:
...
+    )  # Type bytea in the db
+
+    @classmethod
+    def hash_value(
+        cls,
+        value: str,
+        encoding: str = "UTF-8",
+    ) -> str:
+        """Utility function to hash a user's password with a generated salt"""
+        SALT = "$2b$12$UErimNtlsE6qgYf2BrI1Du"
```
Comment (Kelsey-Ethyca) [inline src/fides/api/models/privacy_request.py:1062]: should this be hard coded?

File: src/fides/api/service/connectors/saas_query_config.py [hunk 3]
```diff
@@ -298,6 +298,17 @@ def generate_query(
...
+        custom_privacy_request_fields = input_data.get(CUSTOM_PRIVACY_REQUEST_FIELDS)
+        if (
+            isinstance(custom_privacy_request_fields, list)
+            and len(custom_privacy_request_fields) > 0
+        ):
+            param_values[CUSTOM_PRIVACY_REQUEST_FIELDS] = custom_privacy_request_fields[
+                0
+            ]
+
+        logger.info(param_values)
```
Comment (Kelsey-Ethyca) [inline src/fides/api/service/connectors/saas_query_config.py:310]: do we need this?

File: src/fides/data/sample_project/privacy_center/config/config.json [hunk 4]
```diff
@@ -13,6 +13,16 @@
...
+      },
+      "custom_privacy_request_fields": {
+        "first_name": {
+          "label": "First name",
+          "required": true
+        },
+        "last_name": {
+          "label": "Last name",
+          "required": false
+        }
```
Comment (Kelsey-Ethyca) [inline src/fides/data/sample_project/privacy_center/config/config.json:25]: In future if the env vars for this are false we should not show this in the privacy center

**Review by Kelsey-Ethyca — APPROVED** [review]
Thanks for the walk through. I left some minor comments but its Approved!

File: src/fides/api/service/connectors/saas_query_config.py (hunk 3 above)
Comment (galvana) [reply src/fides/api/service/connectors/saas_query_config.py:310]: No we don't! Removing

**Pull request merged**